types/node/readline/promises.d.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * readline/promises
+ * @since v17.0.0
+ */
+declare module 'readline/promises' {
+    import { EventEmitter } from "events";
+    import { CursorPos, ReadLineOptions, Direction, CompleterResult, Key } from "readline";
+    interface InterfaceEvents {
+        close: () => void;
+        line: (input: string) => void;
+        history: (history: string[]) => void;
+        pause: () => void;
+        resume: () => void;
+        SIGCONT: () => void;
+        SIGINT: () => void;
+        SIGTSTP: () => void;
+    }
+    type Completer = (line: string) => CompleterResult | Promise<CompleterResult>;
+    class Interface extends EventEmitter {
+        readonly terminal: boolean;
+        readonly line: string;
+        readonly cursor: number;
+        protected constructor(input: NodeJS.ReadableStream, output?: NodeJS.WritableStream, completer?: Completer, terminal?: boolean);
```

Comment:
Should we expose those? Is `Interface` designed to be extended?

Reply:
They are exposed in the callback API variant.